api/provider/twitter.go

```diff
@@ -102,19 +112,18 @@ func (t TwitterProvider) FetchUserData(ctx context.Context, tok *oauth.AccessTok
 			Picture:           u.AvatarURL,
 			PreferredUsername: u.UserName,
 			Email:             u.Email,
-			EmailVerified:     true,
+			EmailVerified:     emailVerified,
+			CustomClaims: map[string]interface{}{
+				"needs_phone_verification": u.NeedsPhoneVerification,
```

Comment:
I think requiring any kind of verification based on OAuth values might not be possible. In the case with [snapchat](https://github.com/supabase/gotrue/pull/449), there is no phone or email information. Some might also say that being able to sign in to an OAuth provider counts as a form of verification. 

I think email and phone verification will need to be thought of as a separate process and it should be easy to check if a user is verified or not. Someone using GoTrue could then require users to verify before granting additional permissions.